src/crimson/net/Dispatcher.h

```diff
@@ -36,9 +36,7 @@ class Dispatcher : public boost::intrusive::slist_base_hook<
   // to prevent other dispatchers from processing it, and returns a future
   // to throttle the connection if it's too busy. Else, it returns false and
   // the second future is ignored.
-  virtual std::tuple<bool, seastar::future<>> ms_dispatch(Connection* conn, MessageRef m) {
-    return {false, seastar::now<>()};
-  }
+  virtual std::tuple<bool, seastar::future<>> ms_dispatch(Connection*, MessageRef) = 0;
```

Comment:
after a second though, how about returning `std::optional<future<>>` ? i can help on working on this as a following-up change if you think it's a good idea.

Reply:
Yeah, I think it's a good idea. See https://github.com/ceph/ceph/pull/38360/commits/66d17ffa9cf20d62e473d432f05fd390e71ab4a4